tests/unit/backbeatConsumer.js

```diff
@@ -23,9 +33,169 @@ describe('backbeatConsumer', () => {
         });
         assert.strictEqual(backbeatConsumer._topic, 'testing.my-test-topic');
     });
+});
+
+describe('pause/resume topic partitions on circuit breaker', () => {
+    let consumer;
+
+    beforeEach(() => {
+        consumer = new BackbeatConsumerMock({
+            kafka,
+            groupId: 'unittest-group',
+            topic: 'my-test-topic',
+        });
+
+        const mockConsumer = {
+            pause: sinon.stub().returns(),
+            resume: sinon.stub().returns(),
+            isConnected: () => false,
+        };
+        consumer._consumer = mockConsumer;
+    });
 
     afterEach(() => {
-        process.env.KAFKA_TOPIC_PREFIX = '';
+        sinon.restore();
     });
-});
 
+    describe('_pauseAssignments', () => {
+        it('should not call pause when no paritions assigned', () => {
+            consumer._consumer.assignments = () => [];
+            consumer._pauseAssignments();
+            assert(consumer._consumer.pause.notCalled);
+        });
+
+        it('should pause all assignments', () => {
+            const assignments = [
+                { topic: 'my-test-topic', partition: 0 },
+                { topic: 'my-test-topic', partition: 1 },
+                { topic: 'my-test-topic', partition: 2 },
+            ];
+            consumer._consumer.assignments = () => assignments;
+
+            consumer._pauseAssignments();
+            consumer._pausedPartitions.has('my-test-topic:0');
+            consumer._pausedPartitions.has('my-test-topic:1');
+            consumer._pausedPartitions.has('my-test-topic:2');
```

Comment:
not needed, this is internal state: the "exposed" state is tested already through the mock consumer